numpy/fft/_pocketfft_umath.cpp

```diff
@@ -0,0 +1,434 @@
+/*
+ * This file is part of pocketfft.
+ * Licensed under a 3-clause BSD style license - see LICENSE.md
+ */
+
+/*
+ *  Main implementation file.
+ *
+ *  Copyright (C) 2004-2018 Max-Planck-Society
+ *  \author Martin Reinecke
+ */
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#define PY_SSIZE_T_CLEAN
+#include <assert.h>
+#include <Python.h>
+
+#include "numpy/arrayobject.h"
+#include "numpy/ufuncobject.h"
+
+#include "npy_config.h"
+
```

Comment:
You may want to add an

`#define POCKETFFT_NO_MULTITHREADING`

here. Not sure if it saves much space, but since `numpy` is not going to use parallelism anyway, it cannot hurt.

Reply:
Yes, nice! On the debug build, it reduces the size from 5.1 to 4.7 MB, so helpful.